src/uct/api/uct.h

```diff
@@ -1117,6 +1132,30 @@ struct uct_ep_params {
 };
 
 
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Endpoint attributes, capabilities and limitations.
+ */
+struct uct_ep_attr {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_ep_attr_field. Fields not specified by this mask
+     * will be ignored.
+     */
+    uint64_t                field_mask;
+
+    /**
+     * Local sockaddr used by the endpoint.
+     */
+    struct sockaddr_storage local_address;
+
+    /**
+     * Remote sockaddr the endpoint is connected to.
+     */
+    struct sockaddr_storage remote_address;
```

Comment:
can we do this only for CM endpoint? need to document

Reply:
currently it's only for a CM endpoint but i didn't want to create this limitation in case we would want to expand this later on for 'regular' eps (given that we would port this to the master branch)

Reply:
if we limit this now (in doc) we can always remove this limitation in the future

Reply:
ok so mention in ucp api as well that ucp_ep_create is intended only for the client-server api?

Reply:
in both layers:
- UCP: that endpoint was created with flag XX
- UCT: that endpoint was created with flag YY or flag ZZ